helper_test.go

```diff
@@ -577,6 +577,44 @@ func createWorkspace(t *testing.T, client *Client, org *Organization) (*Workspac
 	}
 }
 
+func createWorkspaceWithVCS(t *testing.T, client *Client, org *Organization) (*Workspace, func()) {
+	var orgCleanup func()
+
+	if org == nil {
+		org, orgCleanup = createOrganization(t, client)
+	}
+
+	ctx := context.Background()
+
+	token, ocTestCleanup := createOAuthToken(t, client, org)
+	w, err := client.Workspaces.Create(ctx, org.Name, WorkspaceCreateOptions{
+		Name: String(randomString(t)),
+		VCSRepo: &VCSRepoOptions{
+			OAuthTokenID: String(token.ID),
+		},
+	})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return w, func() {
+		if err := client.Workspaces.Delete(ctx, org.Name, w.Name); err != nil {
+			t.Errorf("Error destroying workspace! WARNING: Dangling resources\n"+
+				"may exist! The full error is shown below.\n\n"+
+				"Workspace: %s\nError: %s", w.Name, err)
+		}
+
+		if orgCleanup != nil {
+			orgCleanup()
+		}
+
+		if ocTestCleanup != nil {
+			ocTestCleanup()
+		}
```

Comment:
I would swap calling the cleanup functions as the OAuth client is part of the org. So it _should_ be deleted when the org is deleted and so it would error if we try to deleted it after it is already gone.

We could also omit the `ocCleanup()` call, but I like to cleanup the bits in the same order as we created them instead of relying on the last cleanup call to remove everything.